Foundation/NSConcreteValue.swift

```diff
@@ -0,0 +1,199 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+final internal class NSConcreteValue : NSObject, NSCopying, NSSecureCoding, NSCoding {
+    enum SpecialFlags : Int {
+        case NotSpecialType = 0
+    }
+    
+    struct TypeInfoHeader : Equatable {
+        let size : Int
+        let name : String
+
+        init(size: Int, withName name: String) {
+            self.size = size
+            self.name = name
+        }
+    }
+
+    typealias TypeInfo = ManagedBuffer<TypeInfoHeader, Int8>
+    
+    static var _cachedTypeInfo = Dictionary<String, TypeInfo>()
+    
+    var _specialFlags : SpecialFlags = .NotSpecialType
+    var _typeInfo : TypeInfo
+    var _value : UnsafeMutablePointer<UInt8>
+    
+    static func createTypeInfoNonSpecial(spec: String, objCType typep: UnsafePointer<Int8>) -> TypeInfo? {
+        var size: Int = 0
+        var align: Int = 0
+        var count : Int = 0
+        let extraBytes : Int = spec.length + 1 // extra bytes after TypeInfoHeader
+        
+        var type = NSObjCType(spec)
+        guard type != nil else {
+            NSLog("NSConcreteValue.createTypeInfoNonSpecial: unsupported type encoding spec '\(spec)'")
```

Comment:
can't be fatal otherwise you may have a denial-of-service when decoding (unlike similar methods where type is caller supplied, here type is data supplied). will change to print.